spec/controllers/api/v3/patients_controller_spec.rb

```diff
@@ -268,6 +270,51 @@ def create_record_list(n, options = {})
         expect(patient.registration_facility.id).to_not eq current_facility.id
       end
     end
+
+    describe 'soft deletes patients' do
+      before :each do
+        request.env['HTTP_X_USER_ID'] = request_user.id
+        request.env['HTTP_X_FACILITY_ID'] = request_facility.id
+        request.env['HTTP_AUTHORIZATION'] = "Bearer #{request_user.access_token}"
+      end
+
+      let(:existing_patient) { FactoryBot.create(:patient) }
+      let(:deleted_time) { Time.current }
+      let(:delete_patient_payload) do
+        build_patient_payload(existing_patient)
+          .merge(deleted_at: deleted_time,
+                 updated_at: deleted_time,
+                 deleted_reason: 'duplicate')
+      end
+
+      it 'deletes a patient when the patient payload has the deleted_at field set' do
+        expect(Patient.find(existing_patient.id)).to be_present
+        post :sync_from_user, params: { patients: [delete_patient_payload] }, as: :json
+
+        expect(Patient.find_by(id: existing_patient.id)).to be_nil
+        expect(Patient.with_discarded.find_by(id: existing_patient.id)).to be_present
+      end
+
+      it "sets the patient's deleted_reason and deleted_user_id" do
+        expect(Patient.find(existing_patient.id)).to be_present
+        post :sync_from_user, params: { patients: [delete_patient_payload] }, as: :json
+
+        expect(Patient.with_discarded.find_by(id: existing_patient.id).deleted_reason).to eq('duplicate')
+        expect(Patient.with_discarded.find_by(id: existing_patient.id).deleted_by_user_id).to eq(request_user.id)
+      end
+
+      it "doesn't update soft deleted patient's attributes" do
+        existing_patient_name = existing_patient.full_name
+
+        existing_patient.discard_data
+        update_payload_for_discarded_patient = build_patient_payload(existing_patient)
+                                                 .merge(full_name: "Test Patient Name Xcad7asd")
+
+        post :sync_from_user, params: { patients: [update_payload_for_discarded_patient] }, as: :json
+
+        expect(Patient.with_discarded.find_by(id: existing_patient.id).full_name).to eq(existing_patient_name)
+      end
+    end
```

Comment:
We still need a spec here that says something like `expect(patient).to receive(:discard_data)`, right?

Reply:
Done, TIL how to mock this, thanks!